test/format-test.cc

```diff
@@ -1493,8 +1493,13 @@ TEST(format_test, format_pointer) {
   EXPECT_EQ("0x0", fmt::format("{0}", static_cast<void*>(nullptr)));
   EXPECT_EQ("0x1234", fmt::format("{0}", reinterpret_cast<void*>(0x1234)));
   EXPECT_EQ("0x1234", fmt::format("{0:p}", reinterpret_cast<void*>(0x1234)));
+#ifdef __CHERI_PURE_CAPABILITY__
+  EXPECT_EQ("0x" + std::string(sizeof(size_t) * CHAR_BIT / 4, 'f'),
+            fmt::format("{0}", reinterpret_cast<void*>(~uintptr_t())));
```

Comment:
I suggest removing conditional compilation by switching to `size_t` unconditionally as a more portable version. `uintptr_t` should probably be changed to `size_t` as well.

Reply:
And please add a short comment clarifying why we use `size_t` here.

Reply:
You can't use `size_t` instead of `uintptr_t` for the cast, they're different sizes on CHERI, if you want to cast a pointer to a `size_t` it needs to be `static_cast<size_t>(reinterpret_cast<uintptr_t>(ptr))`.

Reply:
> You can't use size_t instead of uintptr_t for the cast

Right, please ignore this part of my comment.

Reply:
Done.